gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -389,4 +417,95 @@ public boolean apply(FileStatus input) {
       return input == null ? false : input.isDir();
     }
   }
+
+  /**
+   * Deploy configs in classpath to HDFS. Finds all the files under
+   * {@link FsDeploymentConfig#getStoreRootNameInClasspath()} in the classpath. For each resource found, creates a
+   * resource on HDFS.
+   *
+   * <p>
+   *  For example: If "test-root" is a resource in classpath and all resources under it needs to be deployed,
+   *  {@link FsDeploymentConfig#getStoreRootNameInClasspath()} is set to "test-root"
+   * <br>
+   * <br>
+   * <b>In Classpath:</b><br>
+   * <blockquote> <code>
+   *       test-root<br>
+   *       &emsp;/data<br>
+   *       &emsp;&emsp;/set1<br>
+   *       &emsp;&emsp;&emsp;/main.conf<br>
+   *       &emsp;/tag<br>
+   *       &emsp;&emsp;/tag1<br>
+   *       &emsp;&emsp;&emsp;/main.conf<br>
+   *     </code> </blockquote>
+   * </p>
+   *
+   * <p>
+   *  A new version 2.0.0 {@link FsDeploymentConfig#getNewVersion()} is created under <code>this.physicalStoreRoot/_CONFIG_STORE</code>
+   *  containing files/directories under "test-root"
+   * <br>
+   * <br>
+   * <b>On HDFS after deploy:</b><br>
+   * <blockquote> <code>
+   *       /_CONFIG_STORE<br>
+   *       &emsp;2.0.0<br>
+   *       &emsp;&emsp;/data<br>
+   *       &emsp;&emsp;&emsp;/set1<br>
+   *       &emsp;&emsp;&emsp;&emsp;/main.conf<br>
+   *       &emsp;&emsp;/tag<br>
+   *       &emsp;&emsp;&emsp;/tag1<br>
+   *       &emsp;&emsp;&emsp;&emsp;/main.conf<br>
+   *     </code> </blockquote>
+   * </p>
+   *
+   */
+  @Override
+  public void deploy(FsDeploymentConfig deploymentConfig) throws IOException {
+
+    log.info("Deploying with config : " + deploymentConfig);
+
+    Path hdfsconfigStoreRoot = new Path(this.physicalStoreRoot.getPath(), CONFIG_STORE_NAME);
+
+    if (!fs.exists(hdfsconfigStoreRoot)) {
+      throw new IOException("Config store root not present at " + this.physicalStoreRoot.getPath());
+    }
+
+    Path hdfsNewVersionPath = new Path(hdfsconfigStoreRoot, deploymentConfig.getNewVersion());
+
+    ConfigurationBuilder cb =
```

Comment:
All the logic for deploying a config store from the classpath is probably re-usable for other `ConfigStore`s I suggest creating an abstract class called `DeployableFromClasspath` that encapsulates all this logic.